bundles/org.openhab.binding.energenie/src/main/java/org/openhab/binding/energenie/internal/handler/EnergenieHandler.java

```diff
@@ -0,0 +1,183 @@
+        } else if (command instanceof OnOffType) {
+            final byte[] ctrl = { DONT_SWITCH, DONT_SWITCH, DONT_SWITCH, DONT_SWITCH };
+            final Matcher matcher = CHANNEL_SOCKET.matcher(channelUID.getId());
+
+            try {
+                boolean found = false;
+
+                if (matcher.find()) {
+                    final int index = Integer.parseInt(matcher.group(1)) - 1;
+
+                    if (index >= 0 && index < SOCKET_COUNT) {
+                        ctrl[index] = OnOffType.ON.equals(command) ? SWITCH_ON : SWITCH_OFF;
+                        stateUpdate(energenieSocket.sendCommand(ctrl));
+                        found = true;
+                    }
+                }
+                if (!found) {
+                    logger.debug("Invalid channel id {}, should be value between 1 and {}", channelUID, SOCKET_COUNT);
+                }
+            } catch (final IOException e) {
+                updateStatus(ThingStatus.OFFLINE);
+                logger.debug("Couldn't get I/O for the connection to: {}:{}.", host, TCP_PORT, e);
+            }
+        }
+    }
+
+    @Override
+    public void initialize() {
+        final EnergenieConfiguration config = getConfigAs(EnergenieConfiguration.class);
+
+        if (config.host != null && config.password != null) {
+            refreshInterval = EnergenieConfiguration.DEFAULT_REFRESH_INTERVAL;
+            host = config.host;
+            logger.debug("Initializing EnergenieHandler for Host '{}'", config.host);
+            energenieSocket = new EnergenieSocket(config.host, config.password);
+
+            updateStatus(ThingStatus.UNKNOWN);
+            refreshJob = scheduler.scheduleWithFixedDelay(this::refreshState, 5, refreshInterval, TimeUnit.SECONDS);
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Can not access device , IP-Address or password not set");
+        }
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("EnergenieHandler disposed.");
+        final ScheduledFuture<?> refreshJob = this.refreshJob;
+
+        if (refreshJob != null && !refreshJob.isCancelled()) {
+            refreshJob.cancel(true);
+            this.refreshJob = null;
+        }
+    }
+
+    private @Nullable Boolean refreshState() {
+        final EnergenieSocket socket = this.energenieSocket;
+
+        if (socket != null) {
+            try {
+                stateUpdate(socket.retrieveStatus());
+                if (thing.getStatus() != ThingStatus.ONLINE && ThingHandlerHelper.isHandlerInitialized(thing)) {
+                    updateStatus(ThingStatus.ONLINE);
+                }
+                return Boolean.TRUE;
+            } catch (final UnknownHostException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Can't find host");
+                logger.debug("Can't find host: {}:{}.", host, TCP_PORT, e);
```

Comment:
is this really needed? The thing status change is logged, so why not
```suggestion
                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Can't find host: " + e.getMessage());
```

Or do you think the full stack trace provides additional information that is helpful in resolving that error? (also applies below)